### PR TITLE
lopper: assists: baremetallinker_xlnx: linker updates

### PR DESCRIPTION
**Pull request by kedareswararao**
1. If the app template has linker_constraints i.e heap and stack size configuration generate the same in cmake meta-data.
2. Update the tcm memory node checks for all the processor instances

### CODE REVIEW

File: lopper/assists/baremetallinker_xlnx.py [hunk 1]
```diff
@@ -177,8 +178,22 @@ def xlnx_generate_bm_linker(tgt_node, sdt, options):
...
+    appname = utils.get_base_name(app_path)
+    yaml_file = os.path.join(app_path, "data", f"{appname}.yaml")
+    stack_size = None
+    heap_size = None
+    if not utils.is_file(yaml_file):
+        print(f"{appname} doesn't have yaml file")
+    else:
+        schema = utils.load_yaml(yaml_file)
+        if schema.get("linker_constraints"):
+            if schema.get("linker_constraints")["stack"]:
```
Comment (onkarharsh) [inline lopper/assists/baremetallinker_xlnx.py:192]: Hi @kedareswararao ,
I guess you intend to use:
    schema["linker_constraints"].get("stack")

And same for heap.

File: lopper/assists/baremetallinker_xlnx.py [hunk 2]
```diff
@@ -282,4 +297,9 @@ def xlnx_generate_bm_linker(tgt_node, sdt, options):
         memip_list.insert(0, memip_list.pop(memip_list.index(has_ocm[0])))
     cfd.write("set(TOTAL_MEM_CONTROLLERS %s)\n" % to_cmakelist(memip_list))
     cfd.write(f'set(MEMORY_SECTION "MEMORY\n{{{mem_sec}\n}}")\n')
+    if stack_size:
+        cfd.write(f'set(STACK_SIZE {stack_size})\n')
```
Comment (onkarharsh) [inline lopper/assists/baremetallinker_xlnx.py:301]: Where are we defining the default size of stack and heap? Is it possible to keep all of them at one place in here?

File: lopper/assists/baremetallinker_xlnx.py (hunk 1 above)
Comment (kedareswararao) [reply lopper/assists/baremetallinker_xlnx.py]: Thanks fixed it now...

File: lopper/assists/baremetallinker_xlnx.py (hunk 2 above)
Comment (kedareswararao) [reply lopper/assists/baremetallinker_xlnx.py:301]: default one we have kept in our linker_gen API...

Comment (kedareswararao) [reply lopper/assists/baremetallinker_xlnx.py]: Fixed now with the latest commit added default stack and heap size based on the processor.

**Pull request merged**